ast/parser.go

```diff
@@ -797,14 +808,60 @@ func (p *Parser) parseExpr() *Expr {
 	return NewExpr(lhs)
 }
 
-// parseTermRelation consumes the next term from the input and returns it. If a
+// parseTermInfixCall consumes the next term from the input and returns it. If a
 // term cannot be parsed the return value is nil and error will be recorded. The
 // scanner will be advanced to the next token before returning.
-func (p *Parser) parseTermRelation() *Term {
-	return p.parseTermRelationRec(nil, p.s.loc.Offset)
+// By starting out with infix relations (==, !=, <, etc) and further calling the
+// other binary operators (|, &, arithmetics), it constitutes the binding
+func (p *Parser) parseTermInfixCall() *Term {
+	return p.parseTermIn(nil, p.s.loc.Offset)
 }
 
-func (p *Parser) parseTermRelationRec(lhs *Term, offset int) *Term {
+func (p *Parser) parseTermIn(lhs *Term, offset int) *Term {
+	// NOTE(sr): `in` is a bit special: besides `lhs in rhs`, it also
+	// supports `key, val in rhs`, so it can have an optional second lhs.
+	if lhs == nil {
+		lhs = p.parseTermRelation(nil, offset)
+	}
+	if lhs != nil {
+		if p.s.tok == tokens.Comma { // second "lhs", or "middle hand side"
+			// NOTE(sr): We need to save and restore here because the way Calls are parsed,
+			// we'd swallow the second argument, attempting to find the mhs of an arity-3
+			// member call.
+			// Concretely, `numbers.range(1,2) in [3]` would be parsed as `member(numbers.range(1), [3])`.
+			s := p.save()
+			p.scan()
+			if mhs := p.parseTermRelation(nil, offset); mhs != nil {
+				if op := p.parseTermOpName(MemberWithKey.Name, tokens.In); op != nil {
+					if rhs := p.parseTermRelation(nil, p.s.loc.Offset); rhs != nil {
+						call := p.setLoc(CallTerm(op, lhs, mhs, rhs), lhs.Location, offset, p.s.lastEnd)
+						switch p.s.tok {
+						case tokens.In:
+							return p.parseTermIn(call, offset)
+						default:
+							return call
+						}
+					}
+				}
+			}
+			p.restore(s)
+		}
```

Comment:
⚠️ I believe the parsing bug is somewhere along these lines. Concretely, `plus(1,2,3,4)` no longer fails in the right way, and the result of something like `plus(1,2,3)` also changed in unexpected ways.